docs/design/specs/Ecma-335-Augments.md

```diff
@@ -810,3 +820,86 @@ Bullet 2: It is valid to call a virtual method using `call` (rather than `callvi
+specially).
+2. For each parameter type of P of T, and corresponding type Q of U, P is identical Q.
+3. For the return type P of T, and return type Q of U, Q is *compatible-with* P and is a *reference type* OR Q is identical to P.
+
+### II.10.3.2 The .override directive (page 147)
+
+Edit the first paragraph "The .override directive specifies that a virtual method shall be implemented (overridden), in this type, by a virtual method with a different name, but with a *covariant-return-compatible-with* signature (§I.8.7.1). This directive can be used to provide an implementation for a virtual method inherited from a base class, or a virtual method specified in an interface implemented by this type and all other. If not used to provide an implementation for a virtual method inherited from a base class the signature must be the identical."
+
+### II.10.3.4 Impact of overrides on derived classes
+Add a third bullet
+"- If a virtual method is overridden via a .override directive and virtual method in parent class was overriden with a .override directive where the method body of that second .override directive is decorated with `System.Runtime.CompilerServices.PreserveBaseOverridesAttribute` then the virtual method override shall apply to the method body of the second .override directive as well.
+
+```
+.class A {
+  .method newslot virtual RetType VirtualFunction() { ... }
+}
+.class B extends A {
+  .method virtual DerivedRetType VirtualFunction() {
+    .custom instance void [System.Runtime]System.Runtime.CompilerServices.PreserveBaseOverridesAttribute::.ctor() = ( 01 00 00 00 ) 
+    .override A.VirtualFuncion 
+    ...
+  }
+}
+.class C extends B {
+  .method virtual MoreDerivedRetType VirtualFunction()
+  {
+    .override A.VirtualFunction
+    ...
+  }
+}
+```
+For this example, the behavior of calls on objects of various types is presented in the following table:
+| Type of object | Method Invocation | Method Called |
+| ---  | --- | --- |
+| A | A::VirtualFunction() | A::VirtualFunction |
+| B | A::VirtualFunction() | B::VirtualFunction |
+| B | B::VirtualFunction() | B::VirtualFunction |
+| C | A::VirtualFunction() | C::VirtualFunction |
+| C | B::VirtualFunction() | C::VirtualFunction |
+| C | C::VirtualFunction() | C::VirtualFunction |
+"
+
+Add a new paragraph "When a rules above result in the implementation of virtual method not satisfying the requirement that the signature of the implementor of a virtual method is not *covariant-return-compatible-with* (§I.8.7.1) the program is not considered well formed. As an example, this can happen if a virtual method is overriden with a .override directive(A) paired with a `PreserveBaseOverridesAttribute` with a *covariant-return-compatible-with* signature. and then the initial virtual method is subsequently overridden with a method which is *covariant-return-compatible-with* the initial virtual method, but not with the method body associated with .override directive(A)."
```

Comment:
A nit "When a rules" -> "When rules"

Reply:
(or maybe *the rules*)

Reply:
(or *a rule* even though I'm not sure whether there's a real difference between the singular and plural in this case)

Reply:
Actually, I think there's also a confusing double negative in the sentence unless I'm reading it wrong - "something results in the implementation of virtual method not satisfying that the signature of something is not covariant compatible", at the very least it's quite a bit to chew on which is the traditional hell of all specs :-).